docs/achitecture_and_design/api_design.md

```diff
@@ -0,0 +1,121 @@
+# API DESIGN
+
+The following are the RESTful endpoints needed to support basic chat and follow-up creation, retrieval, updating, and deletion.
+
+## Chats
+
+### Retrieve individual chat
+`GET /api/chats/<id>`
+```json
+{
+    "id": 2,
+    "user": {
+        "id": 1,
+        "name": "..."
+    },
+    "title": "...",
+    "description": "...",
+    "date": "...",
+    "follow_ups_url": "api/chats/<id>/followups",
+    "follow_up_count": 5
+}
+```
+
+### Retrieve a user's chats
+`GET /api/user/2/chats`
+```json
+{
+    "count": 123,
+    "next": "...url to next page of results...",
+    "previous": "...url to next previous of results...",
+    "chats": [
+        {
+            "id": 1,
+            "user": {
```

Comment:
A single user owns the chat? Is there a M2M for "involved people" and would you expect to encode that here or in the ChatDetail?

Reply:
This comment ended up generating a lot of discussion in our hack tonight regarding both the API and the underlying data model. We're thinking about moving away from `user` and to `participants`.  Where a participant is your M2M relationship between user and chat, but we will probably also have a type associated with the participant which can be CREATOR, INVITEE, ATTENDEE, and maybe COMMENTER. (Can you think of others?)

Reply:
COMMENTER might not be a necessary type since the follow-ups (comments) will have users associated with them.